static/app/components/lazyLoad.tsx

```diff
@@ -39,84 +39,74 @@ type Props<C extends ComponentType> = Omit<React.ComponentProps<C>, 'route'> & {
  *   <LazyLoad component={() => import('./myComponent')} someComponentProps={...} />
  */
 function LazyLoad<C extends ComponentType>(props: Props<C>) {
-  const [LazyComponent, setLazyComponent] = useState<C | null>(null);
-  const [error, setError] = useState<any>(null);
-
-  const handleFetchError = useCallback(
-    (fetchError: any) => {
-      Sentry.withScope(scope => {
-        if (isWebpackChunkLoadingError(fetchError)) {
-          scope.setFingerprint(['webpack', 'error loading chunk']);
+  const importComponent = props.component ?? props.route?.componentPromise;
+
+  const LazyComponent = useMemo(
+    () =>
+      lazy(() => {
+        if (!importComponent) {
+          throw new Error('No component to load');
         }
-        Sentry.captureException(fetchError);
-      });
-
-      // eslint-disable-next-line no-console
-      console.error(fetchError);
-      setError(fetchError);
-    },
-    [setError]
-  );
 
-  const importComponent = props.component ?? props.route?.componentPromise;
+        return retryableImport(importComponent);
+      }),
+    [importComponent]
+  );
 
-  const fetchComponent = useCallback(async () => {
-    if (importComponent === undefined) {
-      return;
-    }
+  return (
+    <ErrorBoundary>
+      <Suspense
+        fallback={
+          <LoadingContainer>
+            <LoadingIndicator />
+          </LoadingContainer>
+        }
+      >
+        <LazyComponent {...(props as React.ComponentProps<C>)} />
+      </Suspense>
+    </ErrorBoundary>
+  );
+}
```

Comment:
```suggestion
  const loadingIndicator = (
    <LoadingContainer>
      <LoadingIndicator />
    </LoadingContainer>
  );

  return (
    <ErrorBoundary>
      <Suspense fallback={loadingIndicator}>
        <LazyComponent {...(props as React.ComponentProps<C>)} />
      </Suspense>
    </ErrorBoundary>
  );
}
```

**complete nit** so feel free to leave it as is, I find this marginally easier to read.

Reply:
I disagree, I don't like having to jump around in the code when reading